File: app/src/main/java/com/jkb/junbin/sharing/feature/message/Message.java
package com.jkb.junbin.sharing.feature.message;

public class Message {
    public int id;
    public String content;
    public long date;
    public String fileName;


    public Message(int id, String content ,String fileName,long date) {
        this.id = id;
        this.content = content;
        this.date = date;
        this.fileName = fileName;
    }
}

File: Feature/message/src/main/java/com/jkb/junbin/sharing/feature/message/Message.kt
package com.jkb.junbin.sharing.feature.message

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.alibaba.android.arouter.launcher.ARouter
import com.jkb.junbin.sharing.feature.message.util.DateUtil
import com.jkb.junbin.sharing.function.shell.interfaces.IFileStatistics

@Entity(tableName = "message_info")
class Message(
    @PrimaryKey @ColumnInfo(name = "id") var id: Int,
    @ColumnInfo(name = "content") var content: String,
    @ColumnInfo(name = "fileName") var fileName: String,
    @ColumnInfo(name = "date") var date: Long
) {
    @Ignore
    val formatDate = DateUtil.getDateToString(date)
    @Ignore
    val downloadCount = ARouter.getInstance().navigation(IFileStatistics::class.java)
        ?.getDownloadCount(id.toString())
}